cmd2/cmd2.py

```diff
@@ -1964,12 +1903,14 @@ def complete_help(self, text, line, begidx, endidx):
             strs_to_match = list(topics | visible_commands)
             matches = self.basic_complete(text, line, begidx, endidx, strs_to_match)
 
-        # Check if we are completing a subcommand
-        elif index == subcmd_index:
-
-            # Match subcommands if any exist
+        # check if the command uses argparser
+        elif index >= subcmd_index and hasattr(self, 'do_' + tokens[cmd_index]) and\
+                hasattr(getattr(self, 'do_' + tokens[cmd_index]), 'has_parser'):
             command = tokens[cmd_index]
-            matches = self.basic_complete(text, line, begidx, endidx, self.get_subcommands(command))
+            cmd_func = getattr(self, 'do_' + command)
+            parser = getattr(cmd_func, 'argparser')
+            completer = AutoCompleter(parser)
+            matches = completer.complete_command_help(tokens[1:], text, line, begidx, endidx)
```

Comment:
Assuming ``has_parser`` can be removed since it appears to always be **True** in an **argparse** command, simplify the code to something like
```Python
elif index >= subcmd_index:
    # If this command uses an argparser, then complete its subcommands if any exist
    try:
        cmd_func = getattr(self, 'do_' + tokens[cmd_index])
        parser = getattr(cmd_func, 'argparser')
        completer = AutoCompleter(parser)
        matches = completer.complete_command_help(tokens[cmd_index:], text, line, begidx, endidx)
    except AttributeError:
        pass
```